codalab/apps/web/static/js/worksheet/worksheet_interface.js

```diff
@@ -143,26 +146,54 @@ var Worksheet = React.createClass({
             return false;
         }.bind(this));
     },
-    toggleEditMode: function(editMode) {
+    toggleEditMode: function(editMode, saveChanges) {
         if (editMode === undefined)
           editMode = !this.state.editMode;  // Toggle by default
 
+        if (saveChanges === undefined)
+            saveChanges = true;
+
         if (!editMode) {
           // Going out of raw mode - save the worksheet.
           if (this.canEdit()) {
-            // TODO: This should be done by the editing control in WorksheetItemList 
-            this.state.ws.info.raw = $("#raw-textarea").val().split('\n');
+            // TODO: This should be done by the editing control in WorksheetItemList
+            var editor = ace.edit('worksheet-editor');
+            if (saveChanges) {
+                this.state.ws.info.raw = editor.getValue().split('\n');
+            }
             this.setState({editMode: editMode});  // Needs to be after getting the raw contents
-            this.saveAndUpdateWorksheet(true);
+            var focusIndex = this.state.ws.info.raw_items_interpreted_items_map[editor.getCursorPosition().row] || [-1,-1];
+            this.saveAndUpdateWorksheet(true, focusIndex[0], focusIndex[1]);
           } else {
             // Not allowed to save worksheet (shouldn't happen).
           }
         } else {
           // Go into edit mode.
           this.setState({editMode: editMode});  // Needs to be before focusing
-          this.setState({activeComponent: 'textarea'});
           // TODO: set cursor intelligently rather than just leaving it at the beginning.
-          $("#raw-textarea").focus();
+          $("#worksheet-editor").focus();
+        }
+    },
+    componentDidUpdate: function() {
+        try {
+            var editor = ace.edit('worksheet-editor');
+            editor.session.setUseWrapMode(false);
+            editor.setShowPrintMargin(false);
+            editor.session.setMode('ace/mode/python');
```

Comment:
changed it to markdown. I think I was testing some python code syntax-highlighting and I forgot to change it to markdown. Thanks for pointing this. :+1: